jena-geosparql/src/main/java/org/apache/jena/geosparql/kryo/NodeSerializer.java

```diff
@@ -0,0 +1,346 @@
+            output.writeString(node.getName());
+        } else if (node.isTripleTerm()) {
+            output.writeByte(TYPE_TRIPLE);
+            kryo.writeObject(output, node.getTriple());
+        } else if (node.isNodeGraph()) {
+            output.writeByte(TYPE_GRAPH);
+            kryo.writeObject(output, node.getGraph());
+        } else if (node.isExt()) {
+            output.writeByte(TYPE_EXT);
+            kryo.writeClassAndObject(output, node);
+        } else {
+            throw new RuntimeException("Unknown node type: " + node);
+        }
+    }
+
+    @Override
+    public Node read(Kryo kryo, Input input, Class<Node> cls) {
+        Node result;
+        String v1, v2;
+        Triple t;
+
+        byte type = input.readByte();
+
+        int typeVal = type & TYPE_MASK;
+        switch (typeVal) {
+            case TYPE_IRI:
+                v1 = input.readString();
+                if ((type & ABBREV_IRI) != 0) {
+                    v1 = decode(prefixToIri, v1);
+                }
+                result = NodeFactory.createURI(v1);
+                break;
+            case TYPE_LITERAL:
+                int subTypeVal = type & LITERAL_MASK;
+                switch (subTypeVal) {
+                    case LITERAL_STRING:
+                        v1 = input.readString();
+                        result = NodeFactory.createLiteralString(v1);
+                        break;
+                    case LITERAL_LANG:
+                        int langDirBits = type & LITERAL_LANG_MASK;
+
+                        TextDirection textDir = switch(langDirBits) {
+                            case LITERAL_LANG_LTR -> TextDirection.LTR;
+                            case LITERAL_LANG_RTL -> TextDirection.RTL;
+                            default -> Node.noTextDirection;
+                        };
+
+                        v1 = input.readString();
+                        v2 = input.readString();
+                        result = NodeFactory.createLiteralDirLang(v1, v2, textDir);
+                        break;
+                    case LITERAL_DTYPE:
+                        v1 = input.readString();
+                        v2 = input.readString();
+                        if ((type & ABBREV_IRI) != 0) {
+                            v2 = decode(prefixToIri, v2);
+                        }
+                        RDFDatatype dtype = typeMapper.getSafeTypeByName(v2);
+                        result = NodeFactory.createLiteralDT(v1, dtype);
+                        break;
+                    default:
+                        throw new RuntimeException("Unknown literal sub-type: " + subTypeVal);
+                }
+                break;
+            case TYPE_BNODE:
+                v1 = input.readString();
+                result = NodeFactory.createBlankNode(v1);
+                break;
+            case TYPE_TRIPLE:
+                t = kryo.readObject(input, Triple.class);
+                result = NodeFactory.createTripleNode(t);
```

Comment:
```suggestion
                result = NodeFactory.createTripleTerm(t);
```
`createTripleNode` is deprecated.